boilerplate/fullstack/remix/app/lib/superTokensHelpers.ts

```diff
@@ -139,3 +146,147 @@ export async function getSessionDetails(
+    req: Request,
+    handler: (error: Error | undefined, session: SessionContainer | undefined) => Promise<Response>,
+    options?: VerifySessionOptions,
+    userContext?: Record<string, any>
+): Promise<Response> {
+    try {
+        const query = getQueryFromRequest(req);
+        const cookies = getCookieFromRequest(req);
+
+        const baseRequest = new PreParsedRequest({
+            method: req.method as HTTPMethod,
+            url: req.url,
+            query: query,
+            headers: req.headers,
+            cookies: cookies,
+            getFormBody: () => req!.formData(),
+            getJSONBody: () => req!.json(),
+        });
+
+        const { session, remixResponse, baseResponse } = await commonSSRSession(
+            baseRequest,
+            options,
+            getUserContext(userContext)
+        );
+
+        if (remixResponse) {
+            return remixResponse as Response;
+        }
+
+        let userResponse: Response;
+
+        try {
+            userResponse = await handler(undefined, session);
+        } catch (err) {
+            await errorHandler()(err, baseRequest, baseResponse, (errorHandlerError: Error) => {
+                if (errorHandlerError) {
+                    throw errorHandlerError;
+                }
+            });
+
+            // The headers in the userResponse are set twice from baseResponse, but the resulting response contains unique headers.
+            userResponse = new Response(baseResponse.body, {
+                status: baseResponse.statusCode,
+                headers: baseResponse.headers,
+            }) as Response;
+
+            let didAddCookies = false;
+            let didAddHeaders = false;
+
+            for (const respCookie of baseResponse.cookies) {
+                didAddCookies = true;
+                userResponse.headers.append(
+                    "Set-Cookie",
+                    serialize(respCookie.key, respCookie.value, {
+                        domain: respCookie.domain,
+                        expires: new Date(respCookie.expires),
+                        httpOnly: respCookie.httpOnly,
+                        path: respCookie.path,
+                        sameSite: respCookie.sameSite,
+                        secure: respCookie.secure,
+                    })
+                );
+            }
+
+            baseResponse.headers.forEach((value: string, key: string) => {
+                didAddHeaders = true;
+                userResponse.headers.set(key, value);
+            });
+            if (didAddCookies || didAddHeaders) {
+                if (!userResponse.headers.has("Cache-Control")) {
+                    // This is needed for production deployments with Vercel
+                    userResponse.headers.set("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
+                }
+            }
+        }
+
+        return userResponse;
+    } catch (error) {
+        console.error(error);
```

Comment:
please no console errors. Would be great if you can double check your PRs before submitting for review for things like this.

Reply:
:heavy_check_mark: